qiita_pet/handlers/api_proxy/prep_template.py

```diff
@@ -624,37 +625,48 @@ def prep_template_graph_get_req(prep_id, user_id):
     # doesn't have full access to the study
     full_access = Study(prep.study_id).can_edit(User(user_id))
 
-    G = prep.artifact.descendants_with_jobs
-
-    # n[0] is the data type: job/artifact
-    # n[1] is the object
-    node_labels = []
-    for n in G.nodes():
-        if n[0] == 'job':
-            atype = 'job'
-            name = n[1].command.name
-        elif n[0] == 'artifact':
-            atype = n[1].artifact_type
-            if full_access or n[1].visibility == 'public':
-                name = '%s\n(%s)' % (n[1].name, n[1].artifact_type)
-            else:
-                continue
-        else:
-            # this should never happen but let's add it just in case
-            raise ValueError('not valid node type: %s' % n[0])
-        node_labels.append((n[0], atype, n[1].id, name))
+    artifact = prep.artifact
+
+    if artifact is None:
+        return {'edges': [], 'nodes': [],
+                'status': 'success', 'message': ''}
+
+    G = artifact.descendants_with_jobs
 
-    return {'edge_list': [(n[1].id, m[1].id) for n, m in G.edges()],
-            'node_labels': node_labels,
+    nodes, edges, wf_id = get_network_nodes_edges(G, full_access)
+
+    return {'edges': edges,
+            'nodes': nodes,
+            'workflow': wf_id,
             'status': 'success',
             'message': ''}
 
-    G = prep.artifact.descendants
-    node_ids = [id_ for id_, label in node_labels]
-    edge_list = [(n.id, m.id) for n, m in G.edges()
-                 if n.id in node_ids and m.id in node_ids]
 
-    return {'status': 'success',
-            'message': '',
-            'edge_list': edge_list,
-            'node_labels': node_labels}
+def prep_template_jobs_get_req(prep_id, user_id):
```

Comment:
Should this check for permission access? Also, currently, user_id is not used in the code

Reply:
Added - also a test for that case